### PR TITLE
Feat 558 header center break titles

### PR DESCRIPTION
**Pull request by yaeluria**
https://github.com/hasadna/anyway-newsflash-infographics/issues/558

### CODE REVIEW

File: src/components/molecules/card/CardHeader.tsx [hunk 1]
```diff
@@ -11,16 +11,26 @@ interface IProps {
...
 }
-const CardHeader: FC<IProps> = ({ variant, text, road }) => {
+const CardHeader: FC<IProps> = ({ variant, road, text }) => {
   let headerContent = null;
+  let textLine1;
+  let textLine2;
+  const textWordsArr = text?.split(' ');
+  const textWordLength = textWordsArr?.length;
+  if (textWordLength && textWordLength > 5) {
+    const indexToSplit = textWordLength / 2;
```
Comment (yuvalbl) [inline src/components/molecules/card/CardHeader.tsx:21]:
```
const indexToSplit = Math.floor(textWordLength / 2);
```

File: src/components/molecules/card/CardHeader.tsx [hunk 2]
```diff
@@ -11,16 +11,26 @@ interface IProps {
   text: string | undefined;
   road: number;
 }
-const CardHeader: FC<IProps> = ({ variant, text, road }) => {
+const CardHeader: FC<IProps> = ({ variant, road, text }) => {
```
Comment (yuvalbl) [inline src/components/molecules/card/CardHeader.tsx:14]: unnecessary change

File: src/components/molecules/card/CardHeader.tsx [hunk 3]
```diff
@@ -11,16 +11,26 @@ interface IProps {
...
   let headerContent = null;
+  let textLine1;
```
Comment (yuvalbl) [inline src/components/molecules/card/CardHeader.tsx:16]: the default should be equal to `text`

File: src/components/molecules/card/CardHeader.tsx [hunk 4]
```diff
@@ -11,16 +11,26 @@ interface IProps {
...
+  }
 
   switch (variant) {
     case HeaderVariant.Centered:
       headerContent = (
         <Box display="flex" alignItems="center" flex={1}>
           <RoadNumberImage roadNumber={road} />
           <Box flex={1} px={2} textAlign="center">
-            <Typography.Body1>{text}</Typography.Body1>
+            <Typography.Body1>{textLine1 ? textLine1 : text}</Typography.Body1>
```
Comment (yuvalbl) [inline src/components/molecules/card/CardHeader.tsx:32]:
```
            <Typography.Body1>{textLine1}</Typography.Body1>
```

File: src/components/molecules/card/CardHeader.tsx [hunk 5]
```diff
@@ -11,16 +11,26 @@ interface IProps {
...
   road: number;
 }
-const CardHeader: FC<IProps> = ({ variant, text, road }) => {
+const CardHeader: FC<IProps> = ({ variant, road, text }) => {
   let headerContent = null;
+  let textLine1;
+  let textLine2;
+  const textWordsArr = text?.split(' ');
+  const textWordLength = textWordsArr?.length;
+  if (textWordLength && textWordLength > 5) {
```
Comment (yuvalbl) [inline src/components/molecules/card/CardHeader.tsx:20]: Avoid magic numbers
https://en.wikipedia.org/wiki/Magic_number_(programming)

**Comment by yaeluria** [comment]
@yuvalbl I've implemented the suggestions in the comments. There still is an issue of edge cases with longer titles. If there are more than 10-12 words (depending on characters) the single lines will break and flow to the next line (wrapping the two lines in flex column is only a partial solution). What should be the behavior then? or maybe we should limit the title length on the server-side?
title when line 1 too long
image

**Comment by yuvalbl** [comment]
@atalyaalon @galraij 
Please see images above

**Comment by yuvalbl** [comment]
Approved, see refactor suggestion in #618
